File: Prototype/src/main/java/me/danetnaverno/editoni/common/block/BlockType.java
package me.danetnaverno.editoni.common.block;

import me.danetnaverno.editoni.common.ResourceLocation;
import me.danetnaverno.editoni.common.render.BlockRenderer;
import me.danetnaverno.editoni.common.render.BlockRendererAir;

public class BlockType
{
    public static final BlockType AIR = new BlockType(new ResourceLocation("common", "air"), new BlockRendererAir());

    public final ResourceLocation id;
    public final BlockRenderer renderer;

    public BlockType(ResourceLocation id, BlockRenderer renderer)
    {
        this.id = id;
        this.renderer = renderer;
    }

    @Override
    public String toString()
    {
        return id.toString();
    }
}

File: Prototype/src/main/java/me/danetnaverno/editoni/common/block/BlockType.kt
package me.danetnaverno.editoni.common.block

import me.danetnaverno.editoni.common.render.BlockRenderer
import me.danetnaverno.editoni.common.render.BlockRendererCube
import me.danetnaverno.editoni.common.texture.Texture
import java.nio.file.Paths

open class BlockType(val id: String, val renderer: BlockRenderer)
{
}

class BlockTypeError(id: String)
    : BlockType(id, BlockRendererCube(
        Texture(Paths.get("data/textures/blocks/error.png")),
        Texture(Paths.get("data/textures/blocks/error.png")),
        Texture(Paths.get("data/textures/blocks/n.png")),
        Texture(Paths.get("data/textures/blocks/w.png")),
        Texture(Paths.get("data/textures/blocks/s.png")),
        Texture(Paths.get("data/textures/blocks/e.png"))
        ))